pkgs/marsnix/read-drvs-from-file.nix

```diff
@@ -0,0 +1,20 @@
+let
+  b = builtins;
+  # read newline separated drv paths from file.
+  text = b.readFile (b.getEnv "DRVS");
+  lines = b.filter (x: x != [] && x != "") (b.split "\n" text);
+
+  # load a .drv file so that the evaluator accepts it as derivation
+  loadDrv = drvFile:
```

Comment:
I believe `builtins.import drvFile` should just as well for your purposes and is less hacky.

Reply:
@sternenseemann I've wanted to do marsnix again but implement it in pure Nix using RFC92.

Reply:
@DavHau may have written it this way to work around a bug in cppnix.

> I believe `builtins.import drvFile` should just as well for your purposes and is less hacky.

It probably should.  However when I tried replacing `loadDrv` with `builtins.import` I managed to crash `nix-daemon`!

```
2023-11-21_00:25:38.46594 nix-daemon: src/libstore/derivations.cc:730: static std::string_view nix::BasicDerivation::nameFromPath(const nix::StorePath&): Assertion `hasSuffix(nameWithSuffix, extension)' failed.
```

Note that the assertion is occurring in `nix-daemon`, not `nix-build`; eval succeeds.